File: consensus/src/main/scala/co/topl/consensus/algebras/EtaCalculationAlgebra.scala
package co.topl.consensus.algebras

import co.topl.models.{Eta, Slot, SlotId}

trait EtaCalculationAlgebra[F[_]] {

  /**
   * Determines the eta value at the requested `childSlot` along the tine containing the `parentSlotId`.  In cases
   * where epochOf(childSlot) =!= epochOf(parentSlotId.slot), a new eta value may be calculated.
   */
  def etaToBe(parentSlotId: SlotId, childSlot: Slot): F[Eta]
}

File: dart/consensus/lib/algebras/eta_calculation_algebra.dart
import 'package:topl_protobuf/consensus/models/slot_data.pb.dart';
import 'package:bifrost_common/models/common.dart';
import 'package:fixnum/fixnum.dart';

abstract class EtaCalculationAlgebra {
  Future<Eta> etaToBe(SlotId parentSlotId, Int64 childSlot);
}
